components/ResultsPage/Estimation.tsx

```diff
@@ -0,0 +1,308 @@
+import { useRouter } from 'next/router'
+import { getTranslations, numberToStringCurrency } from '../../i18n/api'
+import { WebTranslations } from '../../i18n/web'
+import { Language, ResultKey } from '../../utils/api/definitions/enums'
+import {
+  BenefitResult,
+  BenefitResultsObject,
+} from '../../utils/api/definitions/types'
+import { useTranslation } from '../Hooks'
+import { EstimatedTotalItem } from './EstimatedTotalItem'
+
+export const Estimation: React.VFC<{
+  partner
+  resultObject
+  resultArray
+  age
+  maritalStatus
+}> = ({ partner, resultObject, resultArray, age, maritalStatus }) => {
+  const tsln = useTranslation<WebTranslations>()
+  const apiTrans = getTranslations(tsln._language)
+  age = Math.round(age)
+
+  const language = useRouter().locale as Language
+
+  const benefitAge = Object.keys(resultObject)[0]
+  const benefitType = Object.keys(resultObject[Object.keys(resultObject)[0]])
+
+  let estimateIsSame = false
+
+  const benefitObject: BenefitResultsObject =
+    resultObject[Object.keys(resultObject)[0]]
+
+  const resultsArray: BenefitResult[] = Object.keys(
+    resultObject[benefitAge]
+  ).map((value) => resultObject[benefitAge][value])
+
+  let eligible = resultsArray.filter(
+    (result) =>
+      result.eligibility?.result === ResultKey.ELIGIBLE ||
+      result.eligibility?.result === ResultKey.INCOME_DEPENDENT
+  )
+
+  const eligibleTotalAmount = eligible.reduce(
+    (sum, obj) => sum + obj.entitlement.result,
+    0
+  )
+
+  let benefitResultArray: BenefitResult[] = []
+  for (const prop in resultArray) {
+    if (prop) {
+      if (resultArray[prop]) {
+        benefitResultArray.push(
+          resultArray[prop][Object.keys(resultArray[prop])[0]]
+        )
+      }
+    }
+  }
+
+  const isFirstOasGis = () => {
+    let isFirst = false
+
+    for (let i = 0; i < benefitResultArray.length; i++) {
+      const obj = benefitResultArray[i]
+      if (
+        Object.keys(obj[Object.keys(obj)[0]]).includes('oas') &&
+        Object.keys(obj[Object.keys(obj)[0]]).includes('gis')
+      ) {
+        // Check if it's the same object
+        if (JSON.stringify(obj) === JSON.stringify(resultObject)) {
+          isFirst = true
+        }
+        break
+      }
+    }
+    return isFirst
+  }
+
+  const isLastOasGis = () => {
+    let isLast = false
+
+    for (let i = benefitResultArray.length - 1; i >= 0; i--) {
+      const obj = benefitResultArray[i]
+      if (
+        Object.keys(obj[Object.keys(obj)[0]]).includes('oas') &&
+        Object.keys(obj[Object.keys(obj)[0]]).includes('gis')
+      ) {
+        // Check if it's the same object
+        if (JSON.stringify(obj) === JSON.stringify(resultObject)) {
+          isLast = true
+        }
+        break
+      }
+    }
+
+    return isLast
+  }
+
+  //BUILD THE SUMMARY STRINGS FOR EACH BENFIT
+  const buildSummaryString = () => {
```

Comment:
This function is a bit difficult to read with the nested ifs but should be alright. Could try to refactor with an LLM potentially?